src/main/java/io/javarig/generator/StringGenerator.java

```diff
@@ -1,21 +1,29 @@
 package io.javarig.generator;
 
+import com.mifmif.common.regex.Generex;
 import io.javarig.RandomInstanceGenerator;
 import lombok.Getter;
 import lombok.Setter;
-import org.apache.commons.lang3.RandomStringUtils;
 
 import java.lang.reflect.Type;
 
+import static io.javarig.util.Utils.removeUnsupportedRegexCharacters;
+
 @Setter
 @Getter
 public class StringGenerator extends TypeGenerator {
     public StringGenerator(Type type, RandomInstanceGenerator randomInstanceGenerator) {
         super(type, randomInstanceGenerator);
     }
 
+    /**
+     * if minLength < possible regex generation, it will be ignored
+     * anchors (^, $) and backslash (\) are not supported, and will be ignored
+     */
     @Override
     public String generate() {
```

Comment:
I think we should have two methods, one for generating a string with a pattern and one for generating a normal string if there is no pattern

Reply:
that is what config (and more precisely the default config) is used for ( no need for two methods for each generationType one with config and one without)